monkey/monkey_island/cc/services/zero_trust/scoutsuite/data_parsing/rule_parsing.py

```diff
@@ -0,0 +1,28 @@
+from typing import Union
+
+from common.utils.code_utils import get_object_value_by_path
+from common.utils.exceptions import RulePathCreatorNotFound
+from monkey_island.cc.services.zero_trust.scoutsuite.consts.ec2_rules import EC2Rules
+from monkey_island.cc.services.zero_trust.scoutsuite.data_parsing.rule_path_building.rule_path_creators_list import \
+    RULE_PATH_CREATORS_LIST
+
+
+class RuleParser:
+
+    @staticmethod
+    def get_rule_data(scoutsuite_data, rule_name: Union[EC2Rules]):
+        rule_path = RuleParser.get_rule_path(rule_name)
+        return get_object_value_by_path(scoutsuite_data, rule_path)
+
+    @staticmethod
+    def get_rule_path(rule_name: Union[EC2Rules]):
+        creator = RuleParser.get_rule_path_creator(rule_name)
+        return creator.build_rule_path(rule_name)
+
+    @staticmethod
+    def get_rule_path_creator(rule_name: Union[EC2Rules]):
+        for rule_path_creator in RULE_PATH_CREATORS_LIST:
```

Comment:
Lets replace that with a dictionary mapping from rule name to rule path creator

Reply:
What would that achieve? I'd rather not

Reply:
"This implementation isn't efficient." - iterating over 20 elements is negligible and I don't want to introduce new code which needs to be read and maintained but effectively does nothing.

Reply:
No option seems more readable/maintainable, because adding new rule path creator to current architecture requires modifications in 2 files: create new `rule_path_creator.py` and add `rule_path_creator` to a list. If you think your second suggestion is more readable/extendable let me know and I'll try to implement it.

Reply:
Added a hashmap for performance